lib/hyacinth/adapters/digital_object_search_adapter/solr/document_adapter.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+module Hyacinth
+  module Adapters
+    module DigitalObjectSearchAdapter
+      class Solr::DocumentAdapter
+        def solr_document_for(digital_object)
+          solr_document = merge_core_fields_for(digital_object)
+          merge_dynamic_fields_for!(digital_object, solr_document)
+          merge_rights_fields_for!(digital_object, solr_document)
+          solr_document
+        end
+
+        def merge_core_fields_for(digital_object, solr_document = {})
```

Comment:
Are we planning on ever calling this method with a second param that isn't `{}`? If not, is it worth getting rid of the second parameter and renaming it to `core_fields_for`?

Reply:
My goals here were:
1. Principle of least surprise about hash-like APIs
2. Reusability/thread-safety of more stateless wrapping class
3. Reusability across solr "profiles" (item/asset/site)
4. Ease of testing

So this seemed like the most straight-forward way to accomplish those - though I might be influenced by prior art in the Hydra world.

Reply:
👍